ocean_lib/config.py

```diff
@@ -145,21 +177,71 @@ def __init__(self, filename=None, options_dict=None, **kwargs):
             with open(filename) as fp:
                 text = fp.read()
                 self.read_string(text)
-
+        self._handle_and_validate_metadata_cache_uri()
         self._load_environ()
 
+    def _handle_and_validate_metadata_cache_uri(self):
+        metadata_cache_uri = self.get(
+            SECTION_RESOURCES, NAME_METADATA_CACHE_URI, fallback=None
+        )
+        aquarius_url = self.get(SECTION_RESOURCES, NAME_AQUARIUS_URL, fallback=None)
+
+        if aquarius_url:
+            self._logger.warning(
+                f"Config: {SECTION_RESOURCES}.{NAME_AQUARIUS_URL} option is deprecated. "
+                f"Use {SECTION_RESOURCES}.{NAME_METADATA_CACHE_URI} instead."
+            )
+
+        # Fallback to aquarius.url
+        if aquarius_url and metadata_cache_uri == DEFAULT_METADATA_CACHE_URI:
+            self.set(SECTION_RESOURCES, NAME_METADATA_CACHE_URI, aquarius_url)
+            self.remove_option(SECTION_RESOURCES, NAME_AQUARIUS_URL)
+            aquarius_url = None
+
+        if aquarius_url and metadata_cache_uri:
+            raise ValueError(
+                (
+                    f"Both {SECTION_RESOURCES}.{NAME_METADATA_CACHE_URI} and "
+                    f"{SECTION_RESOURCES}.{NAME_AQUARIUS_URL} options are set. "
+                    f"Use only {SECTION_RESOURCES}.{NAME_METADATA_CACHE_URI} because "
+                    f"{SECTION_RESOURCES}.{NAME_AQUARIUS_URL} is deprecated."
+                )
+            )
+
     def _load_environ(self):
-        for option_name, environ_item in environ_names.items():
-            value = os.environ.get(environ_item[0])
+        for option_name, environ_item in environ_names_and_sections.items():
+            if option_name == NAME_METADATA_CACHE_URI:
```

Comment:
this has already been tackled in the specific function. We should treat all aspects in "_handle_and_validate...", or keep something very minimal here. Otherwise this chunk of code dedicated to cache_uris distracts devs from the rest of the functionality.

Reply:
This method (`_load_environ`) is for handling the environment variables.  The `_handle_and_validate_metadata_cache_uri` method is for the special treatment of the `resources.metadata_cache.uri` config option (passed via config file, dict, or string). I believe both methods are necessary.

Reply:
at least this one is duplicated. Let's try to offload some functionality to handle_and_validate...

Reply:
The errors/warnings are different so I think both are needed.  One refers to config options like `resources.metadata_cache.uri` the other refers to env vars like `METADATA_CACHE_URI`.

Reply:
all right. Let's keep it this way :)